src/OrchardCore/OrchardCore.Data.YesSql/DbConnectionValidator.cs

```diff
@@ -72,32 +77,76 @@ public async Task<DbConnectionValidatorResult> ValidateAsync(string databaseProv
         }
 
         var selectBuilder = GetSelectBuilderForDocumentTable(tablePrefix, providerName);
-
         try
         {
             var selectCommand = connection.CreateCommand();
             selectCommand.CommandText = selectBuilder.ToSqlString();
 
             using var result = await selectCommand.ExecuteReaderAsync();
-
-            // at this point the query succeeded and the table exists
-            return DbConnectionValidatorResult.DocumentFound;
+            if (!isDefaultShell)
+            {
+                // The 'Document' table exists.
+                return DbConnectionValidatorResult.DocumentTableFound;
+            }
+
+            var requiredColumnsCount = Enumerable.Range(0, result.FieldCount)
+                .Select(result.GetName)
+                .Where(c => _requiredDocumentTableColumns.Contains(c))
```

Comment:
@jtkech I think here we should add `StringComparer.OrdinalIgnoreCase` to Contains. Some RDBMS has settings to force names to lower case.

Reply:
Done

Reply:
I think it ready to be merged.

Reply:
Yes I will do, just wait a little for now as I did 2 changes in the last hour ;)